builtin/logical/cassandra/backend.go

```diff
@@ -83,7 +83,12 @@ func (b *backend) DB(s logical.Storage) (*gocql.Session, error) {
 		return nil, err
 	}
 
-	return createSession(config, s)
+	session, err := createSession(config, s)
+	//  Store the session in backend for reuse
+	b.session = session
```

Comment:
This needs locking around it. See the implementations of e.g. postgres or mysql.

This also needs to be reset if ResetDB is called, and locks need to be used there too.